File: python/7 kyu/267.py
# https://www.codewars.com/kata/55486cb94c9d3251560000ff/train/python

# Passed

import math

def calculate_ratio(w, h):
    if w == 0 or h == 0:
        raise Exception('You threw an error')
    
    while (greatest_common_denominator:= math.gcd(w, h)) != 1:
        w = int(w/greatest_common_denominator)
        h = int(h/greatest_common_denominator)
        
    ratio_string = f"{w}:{h}"
    return ratio_string

output = calculate_ratio(1024, 768)
print(output)

File: nodejs/7 kyu/267.js
// https://www.codewars.com/kata/5663f5305102699bad000056/train/javascript

// Passed

function mxdiflg(a1, a2) {
  if (a1.length === 0 || a2.length === 0) {
    return -1;
  }

  let computeLengths = (arr) => arr.map((word) => word.length);
  let [a1Lengths, a2Lengths] = [computeLengths(a1), computeLengths(a2)];

  let [a1Max, a1Min, a2Max, a2Min] = [
    Math.max(...a1Lengths),
    Math.min(...a1Lengths),
    Math.max(...a2Lengths),
    Math.min(...a2Lengths),
  ];

  let possibleCombinations = [
    Math.abs(a1Max - a2Min),
    Math.abs(a1Min - a2Min),
    Math.abs(a1Max - a2Max),
    Math.abs(a2Max - a1Min),
  ];

  let result = Math.max(...possibleCombinations);
  return result;
}

const output = mxdiflg(
  [
    "hoqq",
    "bbllkw",
    "oox",
    "ejjuyyy",
    "plmiis",
    "xxxzgpsssa",
    "xxwwkktt",
    "znnnnfqknaz",
    "qqquuhii",
    "dvvvwz",
  ],
  ["cccooommaaqqoxii", "gggqaffhhh", "tttoowwwmmww"],
);
console.log(output);